Support sending of moderation messages

llewelld opened the pull request:
Adds the functionality to send messages to the user when the moderation of their stories change. This relies on the option being enabled in their profile.

The 'mod_message.txt' file contains the text of the message. It has some formatting requirements:

1. The first line is the subject line.
2. All subsequent lines will form the message body.
3. {story}, {profile}, {title} and {status} will be replaced with the URL of the story, the URL of the user's profile, the story title and the new moderation status of the story respectively.

llewelld commented:
Fixes #523.

llewelld commented:
As an aside, the OpenHumans messaging interface is really nice 😄

gedankenstuecke commented on server/apps/main/mod_message.txt:
```diff
@@ -0,0 +1,22 @@
+
+{story}
+
+You can disable these alerts at any time by visiting your profile page:
+
+{profile}
```
I wonder if we should put this a the very end as a kind of email signature (I know the OH stuff wraps it's own signature around it, but maybe worth to put that further down?)

gedankenstuecke commented:
Glad to hear that the API was easy to worth with @llewelld ! 🙏

mastoffel reviewed (commented):
Great job @llewelld, works like a charm for me. I really like that the message is in a separate text file and easy to modify. Just two thoughts:

1) Although it's not super complicated, it might be good to put the format of the message (which you described in the PR) into a docstring somewhere?
2) Currently the user does not get an email when the moderation status is set to "in review". I think this is definitely ok for now , but maybe good to have in mind how we deal with the "in review" status in general. If, for example, the "in review" status is used by moderators to take some time to review the story, it might be good to inform the user that this is happening.

gedankenstuecke reviewed (approved):
This works totally as expected, my only comment is probably an issue on my side rather than with the PR :D

llewelld commented:
Thanks for your really nice suggestions @mastoffel. I've updated the PR with some changes.

> 1. Although it's not super complicated, it might be good to put the format of the message (which you described in the PR) into a docstring somewhere?

This is a very good idea. I've added a docstring to the view entry point. I thought about adding the docstring to the helper function, but since the token substitution happens in the view that felt like the more appropriate place.

> 2. Currently the user does not get an email when the moderation status is set to "in review". I think this is definitely ok for now , but maybe good to have in mind how we deal with the "in review" status in general. If, for example, the "in review" status is used by moderators to take some time to review the story, it might be good to inform the user that this is happening.

If I'm recalling correctly there's been some discussion about how best to handle the "in review" status, and whether to expose this to the story author or not. I'm not certain what the conclusion of that discussion was, but I think the question of whether to send an email is closely linked to this. If anyone can fill me in, I can update the code to reflect the decided behaviour (the changes to the code are simple either way).

llewelld commented on server/apps/main/views.py:
```diff
@@ -538,6 +540,35 @@ def moderate_experience(request, uuid):
+                profile = UserProfile.objects.get(user=request.user)
+                if (profile.comms_review and moderation_prior != moderation_status
+                    and moderation_status in ["approved", "rejected"]):
+                    story_url = "{}/main/view/{}".format(settings.OPENHUMANS_APP_BASE_URL,
```
You will certainly have a better handle on this than I do, but looking at the [django-open-humans docs](https://django-open-humans.readthedocs.io/en/latest/modules/getting-started.html#set-up-settings-py) and your [Open Humans code](https://github.com/OpenHumans/django-open-humans/blob/9665d5bd8fac5aa03469a87f1490412f092c3777/openhumans/helpers.py#L8) makes me think that the base URL is correct. For what it's worth, that's what I have in my `.env` file and I've not noticed any issues with it.

I did look for other non-OpenHumans (i.e. AutSPACEs-specific) setting to use instead, but couldn't see anything. Maybe there's something in django I could use for this though?

In the meantime, I've updated the `.env` templates to remove the path, on the assumption this is correct, but am happy to be corrected on it.

gedankenstuecke replied:
Yeah, I'm pretty sure that our default `.env` template was not correct, thanks for fixing it! 👍

gedankenstuecke reviewed (approved):
Great work @llewelld! I agree on deciding how to use the `in_review` status later on and think this looks good to be merged for now! 👍
